x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/details/index.tsx

```diff
@@ -305,6 +306,10 @@ const RuleDetailsPageComponent: React.FC<DetectionEngineComponentProps> = ({
   const [filterGroup, setFilterGroup] = useState<Status>(FILTER_OPEN);
   const [dataViewOptions, setDataViewOptions] = useState<{ [x: string]: DataViewListItem }>({});
 
+  // load saved query only if rule type === 'saved_query', as other rule types still can have saved_id property that is not used
```

Comment:
Could you please clarify in the comment in what cases this can happen?

Reply:
added comment in https://github.com/elastic/kibana/pull/141747